compiler/match_/instructions/mod.rs

```diff
@@ -268,18 +271,101 @@ impl<ID: IrID + Copy> InstructionAPI<ID> for ConstraintInstruction<ID> {
     }
 }
 
+#[derive(Debug, Clone)]
+pub enum CheckVertex<ID> {
```

Comment:
This will be the executor package version of Vertex so we don't have to pass the already resolved `Label`s around.